prov/efa/src/rxr/rxr_ep.c

```diff
@@ -2596,13 +2663,17 @@ int rxr_endpoint(struct fid_domain *domain, struct fi_info *info,
 	rxr_ep->efa_outstanding_tx_ops = 0;
 	rxr_ep->shm_outstanding_tx_ops = 0;
 
-	ret = efa_cq_open(&efa_domain->util_domain.domain_fid, &cq_attr,
-			  &rxr_ep->rdm_cq, rxr_ep);
-	if (ret)
+	assert(!rxr_ep->ibv_cq_ex);
+
+	rxr_ep->ibv_cq_ex = rxr_ep_create_ibv_cq_ex(rxr_ep, &cq_attr);
```

Comment:
I think it is probably better that this part is moved into `rxr_ep_setup_device_cq`.

Reply:
I need the `err_close_shm_ep` label. This function won't be around in a few weeks anyway, so it doesn't have to be perfect?

Reply:
I see. Then no need to bother.